java/flight/flight-sql/src/main/java/org/apache/arrow/flight/sql/FlightSqlClient.java

```diff
@@ -951,12 +951,11 @@ public static class PreparedStatement implements AutoCloseable {
      *                             {@code PreparedStatement} setters.
      */
     public void setParameters(final VectorSchemaRoot parameterBindingRoot) {
-      if (this.parameterBindingRoot != null) {
-        if (this.parameterBindingRoot.equals(parameterBindingRoot)) {
-          return;
-        }
-        this.parameterBindingRoot.close();
+      if (parameterBindingRoot.equals(this.parameterBindingRoot)) {
```

Comment:
Is this right? Doesn't this do a deep vector comparison when you need a reference comparison here?

Reply:
I was just reorganizing this logic a bit but used the same comparison that was there. It does look like a VSR `equals` method does a deep comparison, and from my understanding of the purpose of this method what we want is a reference check. Happy to make the change!

Reply:
Changed to ref equality.